students/msemerni/hw-4/js/index.js

```diff
@@ -0,0 +1,54 @@
+//// Task 1
+//// Створити функцію для підрахунку кількості голосних англійського алфавіту (a, o, e, i, u, y) в довільному тексті.
+//// Деталі: 
+//// функція повинна отримувати 1 аргумент (текст для пошуку (string)) 
+//// та повертати в консоль кількість голосних, якщо такі є в тексті;
+
+function searchVowels(string) {
+  string = string.toLowerCase();
+  string = [...string];
+  let counter = 0;
+
+  for (let i = 0; i < string.length; i++) {
+    if (string[i] === "a" || string[i] === "o" || string[i] === "e" || string[i] === "i" || string[i] === "u" || string[i] === "y") {
+      counter++;
+    }
+  }
+
+  console.log(counter);
+  return counter;
+}
+
+function searchVowelsByRegExp(string) {
+  const vowels = /[aoeiuy]/g;
+  let vowelsArray = string.toLowerCase().match(vowels);
+
+  if (vowelsArray !== null) {
+    console.log(vowelsArray.length);
+    return vowelsArray.length;
+  } else {
+    console.log(0);
+    return 0
+  };
+}
+
+searchVowels("Hello world");
+searchVowels("js");
+searchVowelsByRegExp("Hello world");
+searchVowelsByRegExp("js");
+
+//// Task 2
+//// Пройти курс по основам RegExp ​​https://regexone.com/.
+//// Практичним завданням будуть (не гуглити!!!) написати регулярний вираз для:
+//// - вілідного номеру мобільного телефон (орієнтовний формат +000 1111 22 33)
+//// - валідного електронну адресу
+//// - визначення кредитні картки (довідкова інфа https://www.paypalobjects.com/en_AU/vhelp/paypalmanager_help/credit_card_numbers.html)
+//// American Express starts with 34, 37 length 15
+//// Mastercard starts with 51-55 length 16
+//// Visa starts with 4 length 16
+
+/^\+\d{3}( )\d{4}(( )\d{2}){2}$/g.test("+000 1111 22 33");
+
+/^\w+([-+.]?\w+)*@\w+([-.]?\w+)*(\.\w{2,6})+$/g.test("misha@gmail.com");
```

Comment:
I would limit digits in the second part of the email
<img width="534" alt="Screen Shot 2022-06-07 at 00 26 35" src="https://user-images.githubusercontent.com/42999342/172259603-5acfaca1-dcc1-4b5d-847b-05710e4016a6.png">